Zend/zend_API.h

```diff
@@ -416,8 +416,10 @@ ZEND_API const char *zend_get_type_by_const(int type);
 #define DLEXPORT
 #endif
 
-#define array_init(arg)				ZVAL_ARR((arg), zend_new_array(0))
-#define array_init_size(arg, size)	ZVAL_ARR((arg), zend_new_array(size))
+#define array_init(arg)						ZVAL_ARR((arg), zend_new_array(0))
+#define array_init_size(arg, size)			ZVAL_ARR((arg), zend_new_array(size))
+#define array_init_assoc(arg)				ZVAL_ARR((arg), zend_new_array_assoc(0))
+#define array_init_assoc_size(arg, size)	ZVAL_ARR((arg), zend_new_array_assoc(size))
```

Comment:
HashTable may be in 3 states - ``packed``, ``map`` and ``uninitialized``. ``array_init()`` creates HashTable in ``unitilaized`` state and it may become ``packed`` or ``map`` depending on the following ``zend_hash_real_init_*`` or update operations. 

``array_init_assoc()`` creates HashTable in ``unitilaized`` state as well, despite of the name.